src/main/src/main/java/org/geoserver/config/GeoServerPropertyOverrideConfigurer.java

```diff
@@ -0,0 +1,29 @@
+package org.geoserver.config;
+
+import java.util.regex.Pattern;
+
+import org.geoserver.config.GeoServerDataDirectory;
+import org.springframework.beans.factory.config.PropertyOverrideConfigurer;
+
+/**
+ * 
+ * Allows the use of ${GEOSERVER_DATA_DIR} inside properties to refer to the data directory, 
+ * irrespective of whether that system property has been set or not.
+ * 
+ * @author niels
+ *
+ */
+public class GeoServerPropertyOverrideConfigurer extends PropertyOverrideConfigurer {
```

Comment:
I thought it might be, it is nothing geofence specific. It allows property override configurers to make use of the data directory.